src/neo4j_genai/schema.py

```diff
@@ -0,0 +1,140 @@
+RETURN {type: nodeLabels, properties: properties} AS output
+"""
+
+REL_QUERY = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE type = "RELATIONSHIP" AND elementType = "node"
+UNWIND other AS other_node
+WITH * WHERE NOT label IN $EXCLUDED_LABELS
+    AND NOT other_node IN $EXCLUDED_LABELS
+RETURN {start: label, type: property, end: toString(other_node)} AS output
+"""
+
+
+def query_database(
+    driver: neo4j.Driver, query: str, params: Optional[dict] = None
+) -> list[dict[str, Any]]:
+    """
+    Queries the database.
+
+    Args:
+        driver (neo4j.Driver):  Neo4j Python driver instance.
+        query (str): The cypher query.
+        params (dict, optional): The query parameters. Defaults to None.
+
+    Returns:
+        list[dict[str, Any]]: the result of the query in json format.
+    """
+    if params is None:
+        params = {}
+    data = driver.execute_query(query, params)
+    return [r.data() for r in data.records]
+
+
+def get_schema(
+    driver: neo4j.Driver,
+) -> str:
+    """
+    Returns the schema of the graph.
+
+    Args:
+        driver (neo4j.Driver): Neo4j Python driver instance.
+
+    Returns:
+        str: the graph schema information in a serialized format.
+    """
+    node_properties = [
+        data["output"]
+        for data in query_database(
+            driver,
+            NODE_PROPERTIES_QUERY,
+            params={"EXCLUDED_LABELS": EXCLUDED_LABELS + [BASE_ENTITY_LABEL]},
+        )
+    ]
+
+    rel_properties = [
+        data["output"]
+        for data in query_database(
+            driver, REL_PROPERTIES_QUERY, params={"EXCLUDED_LABELS": EXCLUDED_RELS}
+        )
+    ]
+    relationships = [
+        data["output"]
+        for data in query_database(
+            driver,
+            REL_QUERY,
+            params={"EXCLUDED_LABELS": EXCLUDED_LABELS + [BASE_ENTITY_LABEL]},
+        )
+    ]
+
+    # Format node properties
+    formatted_node_props = []
+    for element in node_properties:
+        props_str = ", ".join(
+            [f"{prop['property']}: {prop['type']}" for prop in element["properties"]]
+        )
+        formatted_node_props.append(f"{element['labels']} {{{props_str}}}")
+
+    # Format relationship properties
+    formatted_rel_props = []
+    for element in rel_properties:
+        props_str = ", ".join(
+            [f"{prop['property']}: {prop['type']}" for prop in element["properties"]]
+        )
+        formatted_rel_props.append(f"{element['type']} {{{props_str}}}")
+
+    # Format relationships
+    formatted_rels = [
+        f"(:{element['start']})-[:{element['type']}]->(:{element['end']})"
+        for element in relationships
+    ]
+
+    return "\n".join(
+        [
+            "Node properties:",
+            "\n".join(formatted_node_props),
+            "Relationship properties:",
+            "\n".join(formatted_rel_props),
+            "The relationships:",
+            "\n".join(formatted_rels),
+        ]
```

Comment:
Is `get_schema` meant to be used to print out this string? If this will be used by us internally I'm wondering if it's better to return this as something structured e.g. dictionary

Reply:
I was doing that initially, but changed to this format after discussion with @oskarhane 

Langchain supports both formats, maybe it would make sense for us to add a method for returning the schema in a structured format?

Reply:
I could add a `get_structured_schema() -> Dict[str, Any]` method, if that would be valuable.

Reply:
Yeah, I think both would be valuable. We definitely want to have a string output in a format that we think/know the LLM can interpret rather than having the developers come up with their own format.

Reply:
I'll add it in a separate PR, so that we can get this one merged cc @willtai